util/include/Metrics.hpp

```diff
@@ -96,6 +104,15 @@ class Counter {
   uint64_t val_;
 };
 
+// A generic struct that may represent a counter or a gauge
+// the motivation is to eliminate that need to know the exact
+// metric name before getting it from the aggregator
+struct metric_ {
+  std::string component;
+  std::string name;
+  std::variant<Counter, Gauge, Status, concordUtils::Histogram> value;
```

Comment:
Oh sorry, I misspoke in my comment and you took me literally :) I thought we might have added a Histogram type to metrics.  Since we haven't lets just remove concordUtils::Historgram from the variant.

Otherwise, this looks great. Thank you for the change.